core/domain/fs_domain.py

```diff
@@ -108,96 +108,98 @@ def version(self):
+    folder. An asset has no meaning outside its entity, so the assets in
     these asset folders should therefore not be edited directly. They should
     only be modified as side-effects of some other operation (such as adding an
-    image to an exploration).
+    image to an entity like topics, stories or explorations).
 
-    The content of an exploration should include a reference to the asset
+    The content of an entity should include a reference to the asset
     together with the version number of the asset. This allows the
-    exploration to refer to asset versions.
+    entity to refer to asset versions.
 
     In general, assets should be retrieved only within the context of the
-    exploration that contains them, and should not be retrieved outside this
+    entity that contains them, and should not be retrieved outside this
     context.
 
     Args:
-        exploration_id: str. The id of the exploration.
+        assets_path: str. The path to the assets folder of the corresponding
+            entity. It is of the form {{entity}}/{{entity_id}}
+            (eg: exploration/{{exp_id}})
     """
 
     _DEFAULT_VERSION_NUMBER = 1
 
-    def __init__(self, exploration_id):
-        """Constructs a ExplorationFileSystem object.
+    def __init__(self, assets_path):
+        """Constructs a EntityFileSystem object.
 
         Args:
-            exploration_id: str. The id of the exploration.
+            assets_path: str. The path to the parent directory for all assets.
         """
-        self._exploration_id = exploration_id
+        self._assets_path = assets_path
 
     @property
-    def exploration_id(self):
-        """Returns the id of the exploration.
+    def assets_path(self):
+        """Returns the path of the parent folder of assets.
 
         Returns:
-            str. The exploration id.
+            str. The path.
         """
-        return self._exploration_id
+        return self._assets_path
 
     def _get_file_metadata(self, filepath, version):
         """Return the desired file metadata.
 
         Returns None if the file does not exist.
 
         Args:
-            filepath: str. The path to the relevant file within the exploration.
+            filepath: str. The path to the relevant file within the entity.
             version: int. The version number of the file whose metadata is to be
                 returned.
 
         Returns:
             FileMetadataModel or None. The model instance representing the file
-                metadata with the given exploration_id, filepath, and version,
+                metadata with the given assets_path, filepath, and version,
                 or None if the file does not exist.
         """
         if version is None:
             return file_models.FileMetadataModel.get_model(
-                self._exploration_id, 'assets/%s' % filepath)
+                self._assets_path, 'assets/%s' % filepath)
```

Comment:
Just to check ... the reason this isn't an issue is because the local dev store model IDs are transient -- so, although they used to be exp IDs but now they are asset paths, which is still OK as long as the local devserver gets restarted. Is that correct?

Do we need to modify the documentation for FileMetadataModel.get_model()?

Reply:
Yeah, so as long as the initial parameter ends in a directory that has an assets folder, its fine.
We don't need to modify it per se, but I'll change the variable names there from exploration_id to assets_path.

Reply:
Done